### PR TITLE
alsa-lib: 1.1.9 -> 1.2.2 and new alsa conf packages

### PR DESCRIPTION
**Pull request by roastiek**
###### Motivation for this change
alsa-lib is behind upstream and other alsa tools in nixpkgs. This PR adds to new packages `alsa-ucm-conf` and `alsa-topology-conf` to reflect upstream's split to multiple repositories. Their content is linked into alsa-lib `$out/share/alsa` directory, where alsa searches them.

It looks like new alsa-lib is needed by new sof drivers to fully support sound in newer intel notebooks like mine (tested with custom nixos build).
https://bugzilla.redhat.com/show_bug.cgi?id=1772498
https://bugzilla.kernel.org/show_bug.cgi?id=201251

I removed some obsolete seds chaning `u_int` to `uint`. Upstream has fixed this some time ago. And the last remaining is probaly not needed anymore too.

Replaces #74102

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Emantor** [comment]
This should probably target the staging branch.

**Review by Emantor — APPROVED** [review]
Tested on my System, works as expected.

File: pkgs/os-specific/linux/alsa-ucm-conf/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/alsa
+    cp -r ucm ucm2 $out/share/alsa
+
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (Emantor) [inline pkgs/os-specific/linux/alsa-ucm-conf/default.nix]: Also no maintainer.

File: pkgs/os-specific/linux/alsa-topology-conf/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/alsa
+    cp -r topology $out/share/alsa
+
+    runHook postInstall
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (Emantor) [inline pkgs/os-specific/linux/alsa-topology-conf/default.nix]: There is no `maintainer` here, maybe add yourself?

**Review by Emantor — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/alsa-topology-conf/default.nix (hunk 2 above)
Comment (roastiek) [reply pkgs/os-specific/linux/alsa-topology-conf/default.nix]: I was not sure if I should fill myself. But I will give it a try.

**Pull request merged**